far/findfile.cpp

```diff
@@ -3232,9 +3232,6 @@ FindFiles::FindFiles():
 		LastOptions = Options;
 		strLastFindMask = strFindMask;
 		strLastFindStr = strFindStr;
-
-		if (!strFindStr.empty())
-			Editor::SetReplaceMode(false);
```

Comment:
Removing this looks logical, but what was it for?

Reply:
_Global static_ `ReplaceMode` in Editor used to control the behavior of `Shif+F7` / `Alt+F7`: either next find or next replace, _for all Editors!_ So, in the latest versions, after finding some files, all Editors would do "next find." I _guess_ this was needed in the scenario when the user opens a file for edit from the list of found files and presses `Shif+F7` expecting to _find_ the "containing" pattern. If not these lines, the pattern would be replaced with whatever happened to be the last replace string. I vaguely remember that some time ago the list of found files would switch to the existing Editor instance, if the user clicks `F4` on the file which was already being edited. Or at least it would ask the user and _allow_ to switch to the existing instance. Recently, the list of found files only allows to launch the new Editor instance. Once again, this is my educated guess, I may be wrong. If somebody knows, please chime in.

Assuming the above is correct, I removed the global `ReplaceMode` and use the flag stored in the `SearchReplaceDlgParams` instance owned by the Editor. If this breaks somebody's expectations, or I simply screwed something up, I will fix the behavior according to the actual feature request instead of trying to guess what might be useful.

Reply:
Thanks. 

> Or at least it would ask the user and allow to switch to the existing instance. Recently, the list of found files only allows to launch the new Editor instance.

It does allow that when invoked from panels, but not from find file unfortunately, mainly because such editor instance has to be modal. Apparently there were technical issues with modalizing existing windows in the past, I don't know how relevant this limitation is these days.

Reply:
When an Editor is opened from panels, the confusing scenario with unexpected replacement is moot, so we should be fine here.